src/probnum/statespace/sde.py

```diff
@@ -298,6 +350,97 @@ def f(t, y):
 
         return f, y0
 
+    def _setup_vectorized_mde_forward_sqrt(self, initrv, _diffusion=1.0):
+        r"""Set up forward moment differential equations (MDEs) using a square-root
+        implementation. (https://ieeexplore.ieee.org/document/4045974)
+
+        The covariance :math:`P(t)` obeys the Ricatti equation
+
+        .. math::
+            \dot P(t) = G(t)P(t) + P(t)G^\top(t) + L(t)L^\top(t).
+
+        Let :math:`S(t)` be a square-root of :math:`P(t)`, :math:`P(t)` positive definite, then
+
+        .. math::
+            P(t) = S(t)S^\top(t)
+
+        and we get the Riccati-Equation
+
+        .. math::
+            \dot P(t) = G(t)S(t)S^\top(t) + 1/2 \cdot L(t)L^\top(t)S^{-\top}S^\top
+                        + S(t)S^\top(t)G^\top(t) + 1/2 \cdot S(t)S^{-1}(t)L(t)L^\top(t).
+
+        One solution can be found by the square-root :math:`\dot S(t)`
+
+        .. math::
+            \dot S(t) = G(t)S(t) + (A + 1/2 \cdot L(t)L^\top(t))S^{-\top}
+
+        where :math:`A` is an arbitrary symmetric matrix.
+        :math:`A` can be chosen to make S lower-triangular which can be achieved by
+
+        .. math::
+            M(t) = S^{-1}(t)\dot S(t) + \dot S(t)^\top S^{-\top}
+
+        and
+
+        .. math::
+            M(t) = \bar G(t) + \bar G^\top(t) + \bar L(t) \bar L^\top(t)
+
+        and
+
+        .. math::
+            \bar G(t) = S^{-1}(t)G(t)S(t),
+            \bar L(t) = S^{-1}L(t)
+
+        and
+
+        .. math::
+            \dot S(t) = S(t)[M(t)]_{\mathrm{lt}}
+
+        where :math:`\mathrm{lt}` denotes the lower-triangular operator defined by
+
+        .. math::
+            [M(t)]{_{\mathrm{lt}}}_{ij} =
+                \begin{cases}
+                    0 & i < j\\
+                    1/2 m(t)_{ij} & i=j\\
+                    m(t)_{ij} & i > j
+                \end{cases}.
+
+        Compute an ODE vector field that represents the MDEs and is
+        compatible with scipy.solve_ivp.
+        """
+        dim = len(initrv.mean)
+
+        def f(t, y):
+            # Undo vectorization
+            mean, cov_sqrt_flat = y[:dim], y[dim:]
+            cov_sqrt = cov_sqrt_flat.reshape((dim, dim))
```

Comment:
`cov_cholesky` would be better to have here, because it conveys more information than `cov_sqrt`